### PR TITLE
[WIP] #7 - add get package support 

### PR DESCRIPTION
**Pull request by kmanaseryan**
Implemented `getPackage(params)` function, see Python equivalent API [here](https://docs.ckan.org/en/2.8/api/#ckan.logic.action.get.package_show)

Related issue #7 
Related EPIC #4 

Prettier styling has been used on some files, that's why some places you might see no any logic changes just styling.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -24,24 +24,54 @@ class Client {
 
     // Get the JWT token from CkanAuthz, the user should provide the url
     if (url) {
-      const response = await CkanAuthApi.getJWTFromCkanAuthz(url, this.authToken, scope)
+      const response = await CkanAuthApi.getJWTFromCkanAuthz(
```
Comment (rufuspollock) [inline lib/index.js:27]: this is code prettification. It would be easier if this were a separate PR 😄

File: lib/index.js [hunk 2]
```diff
@@ -24,24 +24,54 @@ class Client {
...
-    throw "invalid URL argument"
+    throw 'invalid URL argument'
   }
 
   put(actionType, data) {
     return ActionApi.action(this.api, this.authToken, actionType, data)
   }
 
+  /**
+   * Parameters for get package request
```
Comment (rufuspollock) [inline lib/index.js:42]: Good documentation.

File: lib/index.js [hunk 3]
```diff
@@ -24,24 +24,54 @@ class Client {
...
+   * @typedef {Object} GetPackageParams
+   * @property {string} id - Id or name of the package
+   * @property {boolean} useDefaultSchema - Indicates whether to use default package schema instead of a custom schema defined with an IDatasetForm plugin (default: false)
+   * @property {boolean} includeTracking - Indicates whether to add tracking information to dataset and resources (default: False)
+   */
+
+  /**
+   *
+   * @param {...GetPackageParams} params - Parameters for get package request {@link GetPackageParams}
```
Comment (rufuspollock) [inline lib/index.js:51]: I would have explicit arguments rather than raw params unless some great advantage.

File: lib/index.js [hunk 4]
```diff
@@ -24,24 +24,54 @@ class Client {
...
+   */
+  getPackage(params) {
```
Comment (rufuspollock) [inline lib/index.js:53]: As discussed we don't need this method we can just have `retrieve`

File: test/mocks/getPackageResponse.json [hunk 5]
```diff
@@ -0,0 +1,70 @@
+{
```
Comment (rufuspollock) [inline test/mocks/getPackageResponse.json:1]: Looks good. BTW I'm increasingly thinking we should have a bunch of core mock data somewhere we can reuse over and over again e.g. in github.com/datopian/ckan-mock-data and we submodule it into projects somewhere ... (but this is an aside).

File: test/get.test.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      })
+      return getPackageResponseMock
+    })
+
+  const response = await client.getPackage({
+    id: 'my_dataset',
+  })
+
+  t.is(scope.isDone(), true)
+  t.deepEqual(response, getPackageResponseMock)
```
Comment (rufuspollock) [inline test/get.test.js:46]: The response from `retrieve` would be in frictionless format ...

**Review by rufuspollock — CHANGES_REQUESTED** [review]
As discussed on call this PR misses the point a bit - it is more like an action API call than `retrieve` method we wanted. I suggest closing this PR and doing a new one.

File: lib/index.js (hunk 1 above)
Comment (kmanaseryan) [reply lib/index.js:27]: Yea totally agree, my bad sorry

**Comment by rufuspollock** [comment]
@kmanaseryan i recommend closing this and starting a clean PR ...
